### PR TITLE
Add box.linters styling functions. add check for treesitter dependencies

### PR DESCRIPTION
**Pull request by radbasa**
*Have you read the [Contributing Guidelines](https://github.com/Appsilon/.github/blob/main/CONTRIBUTING.md)?*

Issue #606

## Description
* Adds box::use styling to format_r()
* Adds documentation for box::use styling

## Definition of Done
- [ ] The change is thoroughly documented.
- [ ] The CI passes (`R CMD check`, linter, unit tests, spelling).
- [ ] Any generated files have been updated (e.g. `.Rd` files with `roxygen2::roxygenise()`)

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -24,7 +24,7 @@ Depends:
   R (>= 2.10)
 Imports:
   box (>= 1.1.3),
-  box.linters (>= 0.9.1),
+  box.linters (>= 10.1.0),
```
Comment (jakubnowicki) [inline DESCRIPTION:27]:
```suggestion
  box.linters (>= 0.10.1),
```

File: R/tools.R [hunk 2]
```diff
@@ -140,10 +158,21 @@ rhino_style <- function() {
 #' }
 #' @export
 format_r <- function(paths) {
+  if (!box.linters::is_treesitter_installed()) {
+    cli::cli_abort(
```
Comment (jakubnowicki) [inline R/tools.R:162]: Please change to warning. We should still allow using `format_r` in projects with R <= 4.3.

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # rhino (development version)
 
+* integrate {box.linters} styling functions
+* add compatibility check for `treesitter` and `treesitter.r` dependencies
```
Comment (jakubnowicki) [inline NEWS.md:4]: Please expand (it is related to styling etc.)

**Comment by radbasa** [comment]
@jakubnowicki Will need to go back to `box.linters::style_*`.

`0.10.2` converts:
```R
# View: Shiny modules and related code.
# https://go.appsilon.com/rhino-project-structure
```

to
```R
NA
NA
```

File: R/tools.R [hunk 4]
```diff
@@ -139,11 +164,24 @@ rhino_style <- function() {
...
+    )
+  }
+
   for (path in paths) {
     if (fs::is_dir(path)) {
-      styler::style_dir(path, style = rhino_style)
+      style_box_use(path, exclude_files = exclude_files)
+      styler::style_dir(path, style = rhino_style, exclude_files = exclude_files)
     } else {
+      style_box_use(path, exclude_files = exclude_files)
```
Comment (radbasa) [inline R/tools.R:184]: Support for `exclude_files` added to both `box.linters` styling and `styler` styling.

File: R/dependencies.R [hunk 5]
```diff
@@ -12,6 +12,9 @@ read_dependencies <- function() {
 }
 
 write_dependencies <- function(deps) {
+  if (getRversion() >= 4.3) {
+    deps <- c(deps, "treesitter", "treesitter.r")
+  }
```
Comment (radbasa) [inline R/dependencies.R:17]: Force install treesitter if R >= 4.3

File: R/tools.R [hunk 6]
```diff
@@ -139,11 +164,24 @@ rhino_style <- function() {
 #'   format_r("app/view")
 #' }
 #' @export
-format_r <- function(paths) {
+format_r <- function(paths, exclude_files = c("__init__\\.R")) {
```
Comment (jakubnowicki) [inline R/tools.R:167]: Do we need to exclude `__init__.R`? The recent upgrade that skips files where there is no `box::use` fixes the problem here.

Comment (radbasa) [reply R/tools.R:167]: You're right. We can leave this NULL.

File: DESCRIPTION [hunk 7]
```diff
@@ -19,12 +19,12 @@ BugReports: https://github.com/Appsilon/rhino/issues
...
 Encoding: UTF-8
 Roxygen: list(markdown = TRUE)
-RoxygenNote: 7.3.1
+RoxygenNote: 7.3.2
 Depends:
   R (>= 2.10)
 Imports:
   box (>= 1.1.3),
-  box.linters (>= 0.9.1),
+  box.linters (>= 0.10.2),
```
Comment (jakubnowicki) [inline DESCRIPTION:27]: Please update

File: R/tools.R [hunk 8]
```diff
@@ -122,12 +137,22 @@ rhino_style <- function() {
...
 #' Format R
 #'
-#' Uses the `{styler}` package to automatically format R sources.
+#' Uses the `{styler}` and `{box.linters}` packages to automatically format R sources.
 #'
 #' The code is formatted according to the `styler::tidyverse_style` guide with one adjustment:
 #' spacing around math operators is not modified to avoid conflicts with `box::use()` statements.
 #'
+#' `box.linters::style_*` functions require the `treesitter` and `treesitter.r` packages. These, in
+#' turn, require R >= 4.3.0. `box.linters` provides styling functions specific to the `box::use()`
```
Comment (jakubnowicki) [inline R/tools.R:146]: Please change the order of those sentences. The information that `box.linters` provides styling for `box::use` should go first, followed by the R version requirement.

**Review by jakubnowicki — APPROVED** [review]
LGTM, please resolve my comment and you can merge.

**Pull request merged**